### PR TITLE
Address issue #263, joint_trajectory_controller - wraparoundOffset

### PR DESCRIPTION
**Pull request by graiola**
This PR should address both #263 and #274.
@progtologist , @bponsler Could you try it and see if it solves your issue? Thanks.
Note that this is for kinetic, but you should be able to cherry-pick it for indigo.

### CODE REVIEW

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_segment.h [hunk 1]
```diff
@@ -222,15 +222,15 @@ Scalar wraparoundJointOffset(const Scalar& prev_position,
                              const Scalar& next_position,
                              const bool&   angle_wraparound)
 {
-  // Return value
+   // Return value
```
Comment (bmagyar) [inline joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_segment.h]: There's no reason to change this line

File: joint_trajectory_controller/test/joint_trajectory_segment_test.cpp [hunk 2]
```diff
@@ -108,6 +113,7 @@ TEST(WraparoundOffsetTest, WrappingPositions)
       EXPECT_NEAR(two_pi, wraparound_offset1, EPS);
       EXPECT_NEAR(2.0 * two_pi, wraparound_offset2, EPS);
     }
+
```
Comment (bmagyar) [inline joint_trajectory_controller/test/joint_trajectory_segment_test.cpp]: There's no reason to change this line

File: joint_trajectory_controller/test/joint_trajectory_segment_test.cpp [hunk 3]
```diff
@@ -382,3 +388,4 @@ int main(int argc, char** argv)
   return RUN_ALL_TESTS();
 }
 
+
```
Comment (bmagyar) [inline joint_trajectory_controller/test/joint_trajectory_segment_test.cpp]: There's no reason to change this line

File: joint_trajectory_controller/test/joint_trajectory_segment_test.cpp [hunk 4]
```diff
@@ -87,6 +87,11 @@ TEST(WraparoundOffsetTest, WrappingPositions)
       EXPECT_NEAR(two_pi, wraparound_offset1, EPS);
       EXPECT_NEAR(0.0, wraparound_offset2, EPS);
     }
+
+    // Special case
+    double wraparound_offset1 = wraparoundJointOffset(M_PI/2, -M_PI, true);
+    EXPECT_NEAR(two_pi, wraparound_offset1, EPS);
+
```
Comment (mathias-luedtke) [inline joint_trajectory_controller/test/joint_trajectory_segment_test.cpp]: What about checking `wraparoundJointOffset(-M_PI/2, M_PI, true);` as well?

**Comment by bponsler** [comment]
Thanks for doing this @graiola! I wasn't going to be able to pull this together until later today.

I tested these changes on indigo and they solve the issue I was seeing before. You can see in the below graph that the desired state (red line) no longer has the upward bump seen in #274.

screenshot7

**Comment by graiola** [comment]
@bponsler  no problem! I am happy that the pr solved the problem ;)
@ipa-mdl could you run some tests with your robot/s to validate the fix? Thanks.

**Comment by mathias-luedtke** [comment]
> @ipa-mdl could you run some tests with your robot/s to validate the fix? Thanks.

@graiola: I am sorry, I cannot test it with hardware. I don't have any right now and we don't use wrap-around configs.

**Comment by graiola** [comment]
@progtologist could you give it a shot? Thanks.

**Comment by progtologist** [comment]
@graiola I cannot test it with hardware right now but it runs correctly in simulation so I see no reason why it shouldn't work with hardware as well! Great job!

**Review by bmagyar — CHANGES_REQUESTED** [review]
Ping

**Review by bmagyar — APPROVED** [review]

**Comment by bmagyar** [comment]
We are almost there! ;) 

Could you tell me what's the difference between the two "Added tests for issue #275" commits? They look the same to me. 
Could you remove one of them and the "merge branch blabla" commit as well?

**Comment by graiola** [comment]
I removed the rqt_plot because it was causing problems with travis.

**Comment by bmagyar** [comment]
Great, thanks!

**Pull request merged**